From bd48c6e8b7b71a4149f90a47230f0139d251931f Mon Sep 17 00:00:00 2001
Message-Id: <bd48c6e8b7b71a4149f90a47230f0139d251931f.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Michel Thierry <michel.thierry@intel.com>
Date: Thu, 13 Mar 2014 16:44:32 +0000
Subject: [PATCH 066/312] REVERTME [VPG]: drm/i915/bdw: Define reg address in
 i915_enable_plane_reserved_reg_bit_2

Use hsw registers for gen8+.

Otherwise u32 reg was only initialized for hsw and vlv, causing a kernel
panic at I915_READ(reg).

Bug introduced in commit: f49bd645277bb471503166bc5c80c55f2bdd7cf7,
REVERTME [VPG]: drm/i915: adding VLV specific sprite surface address
register write.

This commit is tagged as REVERTME, because reg is only left undefined
after the commit mentioned above was added. Once the first patch is
reverted, this commit can also be removed.

Issue: AIAF-3656
Tracked-by: AXIA-5404
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
Reviewed-on: https://android.intel.com/171589
Reviewed-by: Bloomfield, Jon <jon.bloomfield@intel.com>

Cherry-picked from embargoed branch.

Signed-off-by: Oscar Mateo <oscar.mateo@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>

Pulled from ABT/kernel-internal to GMin/gmin.

Change-Id: I2c099acc7d91a735f29f8f12bab15cf92f871d8b
Signed-off-by: Oscar Mateo <oscar.mateo@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 11b1f84..607e970 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -13246,7 +13246,7 @@ int i915_enable_plane_reserved_reg_bit_2(struct drm_device *dev, void *data,
 
 	if (!drm_core_check_feature(dev, DRIVER_MODESET))
 		return -ENODEV;
-	if (IS_HASWELL(dev)) {
+	if (IS_HASWELL(dev) || INTEL_INFO(dev)->gen >= 8) {
 		/* crtc_id is not for Plane*/
 		drmmode_obj = drm_mode_object_find(dev, rrb->plane,
 				DRM_MODE_OBJECT_PLANE);
-- 
1.7.9.5

